### PR TITLE
Add readme for treelstm Sentiment example

### PR DESCRIPTION
**Pull request by psyyz10**
## What changes were proposed in this pull request?
Add readme for treelstm Sentiment example

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/treeLSTMSentiment/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+## Summary
+This example shows how to use BigDL train a model on [Standford Treebank
+dataset](https://nlp.stanford.edu/sentiment/index.html) dataset using binary TreeLSTM and [Glove](https://nlp.stanford.edu/projects/glove/)
+word embedding vectors.   Tree-LSTM is a kind of recursive neural networks, which describes in the paper 
+[Improved Semantic Representations From Tree-Structured Long Short-Term Memory Networks](https://arxiv.org/abs/1503.00075)
+ by Kai Sheng Tai, Richard Socher, and Christopher Manning.
+
+The dataset is a corpus of ~10K one-sentence movie reviews from Rotten Tomatoes. Each sentence has been parsed into constituency trees, which is
```
Comment (jason-dai) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/example/treeLSTMSentiment/README.md]: add reference for constituency trees

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/treeLSTMSentiment/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+[constituency-based parse trees](https://en.wikipedia.org/wiki/Parse_tree#Constituency-based_parse_trees), which is
+kind of binary trees with a word at each leaf. After pre-processing, every node has been tagged a label whose range from -2 to 2, representing 
+the sentiment of the word or phrase. The value from -2 to 2 corresponds to highly negative, moderately negative, neutral, moderately positive and
+highly positive respectively. The root of the tree represents the sentiment of the entire sentence.
+
+## Steps to run this example:
+First run the following script
+
+```{r, engine='sh'}
+./fetch_and_preprocess.sh
```
Comment (zhangxiaoli73) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/example/treeLSTMSentiment/README.md]: not .sh file, but .py script.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/treeLSTMSentiment/README.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+The treebank dataset and the Glove word embedding vectors will be downloaded to
+`/tmp/.bigdl/dataset/` directory, after that the treebank will be split into three folders
+corresponding to train, dev, and test in an appropriate format.
+
+Next just run the following command to run the code:
+
+* Spark local:
+
+```{r, engine='sh'}
+     spark-submit --master "local[physical_core_number]" --driver-memory 20g      \
```
Comment (zhangxiaoli73) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/example/treeLSTMSentiment/README.md]: there will be some error.Exception in thread "main" java.io.FileNotFoundException: /tmp/.bigdl/dataset/sst/vocab-cased.txt (No such file or directory)

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/treeLSTMSentiment/fetch_and_preprocess.py [hunk 4]
```diff
@@ -0,0 +1,392 @@
+"""
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/example/treeLSTMSentiment/fetch_and_preprocess.py]: It's wired to put a python file in scala's folder...

Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/example/treeLSTMSentiment/fetch_and_preprocess.py]: This won't be packaged to dist, I think. So how does user run this example with a prebuild BigDL package.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/treeLSTMSentiment/README.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      
+    ```{r, engine='sh'}
+            MASTER=xxx.xxx.xxx.xxx:xxxx
+            spark-submit --master ${MASTER} --driver-memory 20g --executor-memory 10g      \
+                   --total-executor-cores 32 --executor-cores 8                      \
+                   --class com.intel.analytics.bigdl.example.treeLSTMSentiment.Train \
+                   bigdl-VERSION-jar-with-dependencies.jar
+    ```
+        
+    * Yarn client:
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/example/treeLSTMSentiment/README.md]: After you add an uniform hdfs reader, please add an `yarn cluster` command.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/treeLSTMSentiment/fetch_and_preprocess.py (hunk 4 above)
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/example/treeLSTMSentiment/fetch_and_preprocess.py]: Please add a `fileSet` to `https://github.com/intel-analytics/BigDL/blob/master/spark/dist/assembly/dist.xml`, to copy this python file to dist/bin

**Comment by qiuxin2012** [comment]
LGTM

**Pull request merged**